locker/memory_locker.go

```diff
@@ -0,0 +1,125 @@
+package locker
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type MemoryKey struct {
+	id             string
+	key            string
+	expirationTime time.Time
+	Channel        *chan error
+}
+
+func (k *MemoryKey) ID() string {
+	return k.id
+}
+
+func (k *MemoryKey) Key() string {
+	return k.key
+}
+
+func (k *MemoryKey) ExpirationTime() time.Time {
+	return k.expirationTime
+}
+
+func (k *MemoryKey) SetExpirationTime(t time.Time) error {
+	k.expirationTime = t
+	return nil
+}
+
+type MemoryLocker struct {
+	lockedItems sync.Map
+	mutex       *sync.Mutex
+}
+
+func (m *MemoryLocker) Lock(key string) (Key, error) {
+	if key == "" {
+		return &MemoryKey{}, fmt.Errorf("key is empty")
+	}
+
+	id := uuid.New().String()
+
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if lockInfo, ok := m.lockedItems.Load(key); ok {
+		keyLock := lockInfo.(LockInformation)
+		if time.Since(keyLock.Date) < ValidTimePeriod {
+			return &MemoryKey{}, fmt.Errorf("key '%s' is already locked by: %s", key, keyLock.ID)
+		}
+	}
+
+	lockChannel := make(chan error)
```

Comment:
looks like the `done` pattern, I'd change the name to `doneChannel` or just `done` for clarity.